### PR TITLE
added player functionality

### PR DESCRIPTION
**Pull request by rkapoor10**
Single Video Listing Page added - 
ss

### CODE REVIEW

**Review by rushikesh3299 — APPROVED** [review]
Overall looks good to me.
Just remove console logs & commented code if you don't want it in future

File: src/components/videosContainer/VideosContainer.jsx [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
-        </div>
-    )
-}
+const VideosContainer = ({ heading, videosData }) => {
+  return (
+    <div className="videos-container">
+      <div className="txt-m">{heading}</div>
+      <div className="grid-column-layout">
+        {videosData.map((videoDetails) => (
+          <VideoCard details={videoDetails} />
```
Comment (JantuDeb) [inline src/components/videosContainer/VideosContainer.jsx:11]: add a `key` attribute

File: src/context/VideoContext/VideoContext.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    const [videosData, setVideosData] = useState([])
+    const fetchVideo = async ()=>{
+        try{
+            const {data} = await axios.get('/api/videos')
+            setVideosData(data.videos)
+        }
+        catch(error){
+            console.log("Error in fetching data")
+        }
+    }
```
Comment (JantuDeb) [inline src/context/VideoContext/VideoContext.jsx:19]: You can define this function inside `useEffect`, then there will be no warnings for missing dependency

File: src/pages/VideoListingPage.jsx [hunk 3]
```diff
@@ -1,29 +1,17 @@
...
-        <VideoCard />
-        <VideoCard />
-        <VideoCard />
-        <VideoCard />
-        <VideoCard />
-        <VideoCard />
-        <VideoCard />
-        <VideoCard />
+        {videosData.map((videoDetails)=>
+            <VideoCard details={videoDetails}/>
```
Comment (JantuDeb) [inline src/pages/VideoListingPage.jsx:13]: same here add a `key` attribute

**Review by JantuDeb — APPROVED** [review]
Code looks clean and structured. Go through the comments I added

File: src/components/videosContainer/VideosContainer.jsx (hunk 1 above)
Comment (rkapoor10) [reply src/components/videosContainer/VideosContainer.jsx:11]: done

File: src/context/VideoContext/VideoContext.jsx (hunk 2 above)
Comment (rkapoor10) [reply src/context/VideoContext/VideoContext.jsx:19]: warning is not coming anymore

File: src/pages/VideoListingPage.jsx (hunk 3 above)
Comment (rkapoor10) [reply src/pages/VideoListingPage.jsx:13]: done

**Pull request merged**